integration_tests/pages/shared/caseList.ts

```diff
@@ -44,24 +52,90 @@ export default class CaseListPage extends Page {
   }
 
   shouldContainTableOfReferralSummaries() {
+    this.columnHeaders.forEach((columnHeader, columnHeaderIndex) => {
+      cy.get('.govuk-table__header').eq(columnHeaderIndex).should('have.text', columnHeader)
+    })
+
     cy.get('.govuk-table__body').within(() => {
       cy.get('.govuk-table__row').each((tableRowElement, tableRowElementIndex) => {
         const summary = this.referralSummaries[tableRowElementIndex]
 
         cy.wrap(tableRowElement).within(() => {
-          cy.get('.govuk-table__cell:first-of-type').should(
-            'have.html',
-            `<a class="govuk-link" href="${assessPaths.show.personalDetails({ referralId: summary.id })}">${
-              summary.prisonNumber
-            }</a>`,
-          )
-          cy.get('.govuk-table__cell:nth-of-type(2)').should(
-            'have.text',
-            summary.submittedOn ? DateUtils.govukFormattedFullDateString(summary.submittedOn) : 'N/A',
-          )
-          cy.get('.govuk-table__cell:nth-of-type(3)').should('have.text', summary.courseName)
-          cy.get('.govuk-table__cell:nth-of-type(4)').should('have.text', summary.audiences.join(', '))
-          cy.get('.govuk-table__cell:nth-of-type(5)').should('have.html', CaseListUtils.statusTagHtml(summary.status))
+          this.columnHeaders.forEach((column, columnIndex) => {
+            cy.get('.govuk-table__cell')
+              .eq(columnIndex)
+              .then(tableCellElement => {
+                switch (column) {
+                  case 'Conditional release date':
+                    cy.wrap(tableCellElement).should(
+                      'have.text',
+                      CaseListUtils.tableRowContent(summary, 'Conditional release date'),
```

Comment:
As I've said in a few PR comments, I'm not 100% confident on reusing implementing functions like `tableRowContent` here in the tests. If the implementation of that function changes, we won't catch it in our integration tests, although we _will_ catch it in our unit tests. I don't think this needs to block this PR, but I wonder if when we add the Refer tests we should split this `shared` test out into two tests and handwrite the different columns etc?

Reply:
We could, but we might end up needing to rewrite quite a bit of code every time we use it. We did use `CaseListUtils.statusTagHtml(summary.status)` here before. Maybe one for later if we want to revise our approach across the board in the integration tests

Reply:
We'd also need to do that for every row, but I suppose we can just use a small number of rows in the integration tests when not testing pagination

Reply:
Yeah, I think this is ok for now, but when we split out the two pages we should write it out to make our tests a bit more robust

Reply:
Might be worth a quick huddle if you have a minute later just to make sure I know what you're suggesting before I write the Refer test!